src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -245,12 +300,70 @@ template void CountryInstance::add_leader(LeaderBranched<UnitType::branch_t::NAV
 template bool CountryInstance::remove_leader(LeaderBranched<UnitType::branch_t::LAND> const*);
 template bool CountryInstance::remove_leader(LeaderBranched<UnitType::branch_t::NAVAL> const*);
 
-bool CountryInstance::apply_history_to_country(CountryHistoryEntry const* entry, MapInstance& map_instance) {
-	if (entry == nullptr) {
-		Logger::error("Trying to apply null country history to ", get_identifier());
-		return false;
+template<UnitType::branch_t Branch>
+void CountryInstance::unlock_unit_type(UnitTypeBranched<Branch> const& unit_type) {
+	IndexedMap<UnitTypeBranched<Branch>, bool>& unlocked_unit_types = get_unlocked_unit_types<Branch>();
+
+	decltype(unlocked_regiment_types)::value_ref_t unlock_value = unlocked_unit_types[unit_type];
+
+	if (unlock_value) {
+		Logger::warning(
+			"Attempted to unlock already-unlocked unit type \"", unit_type.get_identifier(),
+			"\" for country ", get_identifier()
+		);
+		return;
 	}
 
+	unlock_value = true;
+
+	if constexpr (Branch == UnitType::branch_t::LAND) {
+		allowed_regiment_cultures = RegimentType::get_more_permissive_allowed_cultures(
+			allowed_regiment_cultures, unit_type.get_allowed_cultures()
+		);
+	}
+}
+
+template void CountryInstance::unlock_unit_type(UnitTypeBranched<UnitType::branch_t::LAND> const&);
+template void CountryInstance::unlock_unit_type(UnitTypeBranched<UnitType::branch_t::NAVAL> const&);
+
+bool CountryInstance::is_primary_culture(Culture const& culture) const {
+	return &culture == primary_culture;
+}
+
+bool CountryInstance::is_accepted_culture(Culture const& culture) const {
+	return accepted_cultures.contains(&culture);
+}
+
+bool CountryInstance::can_recruit_culture(Culture const& culture) const {
```

Comment:
This feels misplaced as recruitment requires the same logic.
Shouldn't this be part of the allowed_cultures_t?
allowed_cultures_t seems like a stategy to me.

Something like allowed_cultures_t.can_recruit(country instance, pop)

Reply:
Good point, I'll move it alongside the `get_more_permissive_allowed_cultures` function